File: app/src/main/java/me/larikraun/tourreviews/network/ReviewRepository.java
package me.larikraun.tourreviews.network;

import javax.inject.Inject;

import io.reactivex.Observable;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;
import me.larikraun.tourreviews.model.ReviewResponse;
import retrofit2.Retrofit;

/**
 * Author: Omolara Adejuwon
 * Date: 17/10/2018.
 */
public class ReviewRepository {
	
	Retrofit mRetrofit;
	
	@Inject
	public ReviewRepository (Retrofit retrofit) {
		mRetrofit = retrofit;
	}
	
	public Observable<ReviewResponse> fetchReviews () {
		return mRetrofit.create (FetchReviewService.class).fetchReviews (4, 0,0,"date_of_review","DESC").subscribeOn (Schedulers.io ()).observeOn (AndroidSchedulers.mainThread ());
	}
	
}

File: app/src/main/java/me/larikraun/tourreviews/network/ReviewRepository.kt
package me.larikraun.tourreviews.network

import io.reactivex.Observable
import me.larikraun.tourreviews.model.ReviewResponse
import retrofit2.Retrofit
import javax.inject.Inject

/**
 * Author: Omolara Adejuwon
 * Date: 17/10/2018.
 */
class ReviewRepository @Inject
constructor(internal var mRetrofit: Retrofit) {

    fun fetchReviews(count: Int, page: Int): Observable<ReviewResponse> {
        return mRetrofit.create(FetchReviewService::class.java).fetchReviews(count, page, 0, "date_of_review", "DESC")
    }

}
